jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/PluginConfigurationProcessor.java

```diff
@@ -249,14 +253,32 @@ static ImageReference parseImageReference(String image, String type) {
     }
 
     if (isWarPackaging(jibPluginConfiguration)) {
-      return JavaEntrypointConstructor.makeDistrolessJettyEntrypoint();
+      return null;
     }
 
     String mainClass = projectProperties.getMainClass(jibPluginConfiguration);
     return JavaEntrypointConstructor.makeDefaultEntrypoint(
         getAppRootChecked(jibPluginConfiguration), jibPluginConfiguration.getJvmFlags(), mainClass);
   }
 
+  /**
+   * Compute the container program arguments. If entrypoint is inherited (null), program arguments
+   * must be inherited (null) if empty (used for Tomcat base images).
```

Comment:
This special handling for Tomcat and the inconsistency between the entrypoint being null and the program arguments being empty bother me. Ideally, I think the entrypoint and the program arguments could be nullable and start with null at the beginning instead of the current defaulting to empty, indicating Jib will by default inherit entrypoint and cmd. Then for non-WAR projects, we are already setting the entrypoint and the cmd explicitly, so we still won't inherit them for the non-WAR projects as usual. That is, the non-WAR is a special case that overrides the default. On the other hand, for WAR, we would not touch the null entrypoint and cmd, and inheriting them would be just the default behavior. Also, I imagine fixing #924 fits naturally in this model, since we can later add an option to not override them in the non-WAR case (through some configuration or so). @GoogleContainerTools/java-tools Thoughts?

Reply:
I agree - we should have the default be `null` and set to empty if needed (so non-WAR would set entrypoint as generated and program arguments to empty).

Reply:
I have a problem. If the user set the ENTRYPOINT to `[catalina.sh, run]` the CMD is null so it will be inherited and we will ENTRYPOINT and CMD equal to `[catalina.sh, run]`.

Can I set CMD to an empty list if the ENTRYPOINT is not null ?

Reply:
I think it's right to not inherit CMD when ENTRYPOINT is redefined: https://github.com/GoogleContainerTools/jib/pull/1071#discussion_r223431180

I'm just wondering if `CMD []` in `Dockerfile` will set the value of the CMD metadata to null or an empty string. I'll have to double-check.

Reply:
> Can I set CMD to an empty list if the ENTRYPOINT is not null ?

Hmm... this actually is tricky.

The value of an empty list is supposed to mean `CMD []`, which will specifically define CMD (and override the base value) to be an empty string. And we are trying to set the behavior here that the null value means inheriting CMD from the base image. That said, there is way to explicitly set CMD to null in the container metadata. Sure, setting to an empty list might be a feasible choice, but this diverges from how Docker build actually works, although I'm not saying we always have to stick to how Docker build works.

Reply:
> there is way to explicitly set CMD to null

Mistyped: there is *no* way to ...